_posts/2023-02-11-new-repo-fastly.markdown

```diff
@@ -0,0 +1,123 @@
+Some synchronize very frequently, as frequently as every 15 minutes.
+Some synchronize more slowly, only once a day.  And some mirrors often
+appear broken and don't appear to have synced at all.  How can we
+possibly know this?  Well we drop a file into the root of the mirror
+at the Shadow level with a timestamp.  By watching the value in this
+file, we can tell how old the contents of any downstream repo are.
+All this information is exposed in our [mirror
+dashboard](https://grafana.s.voidlinux.org/d/cLraC-XMk/mirrors-status).
+The "Origin" referenced in that dashboard is a slight misnomer, its
+actually showing the time behind `now`.
+
+Having multiple mirrors is great, but ensuring that a mirror nearby to
+a user who wants to update, and make sure they've configured and are
+using it is a bit harder.  What we really want is a proxy that will
+always find the nearest mirror and fetch packages from that location
+for a user.  Long time users of Void may remember `auto.voidlinux.org`
+which was an early attempt at this.  Due to limitations in XBPS
+itself, and limitations in the way that we are able to introspect
+mirrors, this service was eventually retracted.  What we actually want
+is a network of servers everywhere well connected to networks that
+users are on, and then with a good connection back to at least one of
+the Void operated mirrors.  This, as you might have guessed, is a CDN.
+
+As mentioned in the [Fast
+Forward](/news/2023/01/fastly-fast-forward.html) post, Void has
+generously been sponsored by Fastly, which has allowed us to provision
+a new repository URL that leverages's Fastly's global network to
+provide what should be the generally optimal mirror regardless of
+where in the world any particular end user exists.  The Fastly mirror
+does not sync, nor does it have a concept of file storage that we
+manage as the project.  It is for all intents and purposes an
+extremely large high speed cache, and when you request a package from
+Fastly, the package will either be returned from one of Fastly's
+servers, or it will be streamed to you via the Fastly network while
+being simulataneously fetched from one of Void's backend servers.  The
+very first download of a package may be slower than if you downloaded
+it from Void directly, but all subsequent fetches will be extremely
+fast.  Since the package distribution follows a sort of curve of
+popularity ranging from the common packages that make up desktop
+environments, the base system, and popular software suites down to
+less commonly installed packages and rarely updated libraries, the
+cache will usually be hot for the most commonly installed packages!
+
+So to recap, packages are built on servers, copied to a central
+server, signed, copied to a shadow mirror, copied to a tier 1 mirror,
+possibly copied to downstream mirrors, and then installed on end
+systems.  Here's a graphic:
+
+![diagram of the package flow](/assets/img/mirrors.svg)
```

Comment:
maybe we should use

```html
<img src="/assets/img/mirrors.svg" alt="diagram of the package flow" style="max-height: 80vh" />
```
as it's kinda huge as-is. that would make it maximally 80% of the viewport height